datastore/src/main/java/org/jboss/pnc/datastore/repositories/internal/ArtifactSpringRepository.java

```diff
@@ -18,12 +18,108 @@
+                    "  ON dependency_artifact.dependency_artifact_id = artifact.id " +
+                    "INNER JOIN BuildRecord buildRecord " +
+                    "  ON dependency_artifact.build_record_id = buildRecord.id " +
+                    "INNER JOIN TargetRepository targetRepository " +
+                    "  ON targetRepository.id = artifact.targetRepository_id " +
+                    "WHERE " +
+                    " buildRecord.id = ?1",
+            nativeQuery = true)
+    Object[] countMinimizedDependencyArtifactsForBuildRecord(Integer buildRecordId);
+
+    @Query(
+            value = "SELECT DISTINCT " +
+                    " artifact.id, " +
+                    " artifact.artifactQuality, " +
+                    " artifact.deployPath, " +
+                    " artifact.filename, " +
+                    " artifact.identifier, " +
+                    " artifact.importDate, " +
+                    " artifact.md5, " +
+                    " artifact.originUrl, " +
+                    " artifact.sha1, " +
+                    " artifact.sha256, " +
+                    " artifact.size, " +
+                    " artifact.targetRepository_id as targetRepositoryId, " +
+                    " targetRepository.temporaryRepo, " +
+                    " targetRepository.identifier as targetRepositoryIdentifier, " +
+                    " targetRepository.repositoryPath, " +
+                    " targetRepository.repositoryType " +
+                    "FROM Artifact artifact " +
+                    "INNER JOIN build_record_built_artifact_map built_artifact " +
+                    "  ON built_artifact.built_artifact_id = artifact.id " +
+                    "INNER JOIN BuildRecord buildRecord " +
+                    "  ON built_artifact.build_record_id = buildRecord.id " +
+                    "INNER JOIN TargetRepository targetRepository " +
+                    "  ON targetRepository.id = artifact.targetRepository_id " +
+                    "WHERE " +
+                    " buildRecord.id = ?1 " +
+                    "ORDER BY " +
+                    " artifact.id ASC LIMIT ?2 OFFSET ?3",
+            nativeQuery = true)
+    List<RawArtifact> getMinimizedBuiltArtifactsForBuildRecord(Integer buildRecordId, int pageSize, int offset);
+
+    @Query(
+            value = "SELECT COUNT(DISTINCT artifact.id) " +
+                    "FROM Artifact artifact " +
+                    "INNER JOIN build_record_built_artifact_map built_artifact " +
+                    "  ON built_artifact.built_artifact_id = artifact.id " +
+                    "INNER JOIN BuildRecord buildRecord " +
+                    "  ON built_artifact.build_record_id = buildRecord.id " +
+                    "INNER JOIN TargetRepository targetRepository " +
+                    "  ON targetRepository.id = artifact.targetRepository_id " +
+                    "WHERE " +
+                    " buildRecord.id = ?1",
+            nativeQuery = true)
```

Comment:
What is the reason to set it as nativeQuery?

Reply:
With standard JPA I could not mange to override at runtime some EAGER fetches, so I had to go to native queries.